File: pageindicatorview/src/main/java/com/rd/draw/drawer/type/SlideDrawer.java
package com.rd.draw.drawer.type;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.support.annotation.NonNull;

import com.rd.animation.data.Value;
import com.rd.animation.data.type.SlideAnimationValue;
import com.rd.draw.data.Indicator;
import com.rd.draw.data.Orientation;

public class SlideDrawer extends BaseDrawer {

    public SlideDrawer(@NonNull Paint paint, @NonNull Indicator indicator) {
        super(paint, indicator);
    }

    public void draw(
            @NonNull Canvas canvas,
            @NonNull Value value,
            int coordinateX,
            int coordinateY) {

        if (!(value instanceof SlideAnimationValue)) {
            return;
        }

        SlideAnimationValue v = (SlideAnimationValue) value;
        int coordinate = v.getCoordinate();
        int unselectedColor = indicator.getUnselectedColor();
        int selectedColor = indicator.getSelectedColor();

        paint.setColor(unselectedColor);
        drawIndicator(canvas,paint,coordinateX,coordinateY);


        paint.setColor(selectedColor);
        int slideX = coordinate;
        int slideY = coordinateY;
        if (indicator.getOrientation() == Orientation.VERTICAL) {
            slideX = coordinateX;
            slideY = coordinate;
        }

        drawIndicator(canvas,paint,slideX,slideY);
    }
}

File: pageindicatorview/src/main/java/com/rd/draw/drawer/type/SlideDrawer.kt
package com.rd.draw.drawer.type

import android.graphics.Canvas
import android.graphics.Paint
import com.rd.animation.data.Value
import com.rd.animation.data.type.SlideAnimationValue
import com.rd.draw.data.Indicator
import com.rd.draw.data.Orientation

class SlideDrawer(paint: Paint, indicator: Indicator) : BaseDrawer(paint, indicator) {

	fun draw(
			canvas: Canvas,
			value: Value,
			coordinateX: Int,
			coordinateY: Int) {

		if (value !is SlideAnimationValue) {
			return
		}

		val coordinate = value.coordinate
		val unselectedColor = indicator.unselectedColor
		val selectedColor = indicator.selectedColor
		val radius = indicator.radius

		paint.color = unselectedColor
		canvas.drawCircle(coordinateX.toFloat(), coordinateY.toFloat(), radius.toFloat(), paint)

		paint.color = selectedColor
		if (indicator.orientation === Orientation.HORIZONTAL) {
			canvas.drawCircle(coordinate.toFloat(), coordinateY.toFloat(), radius.toFloat(), paint)
		} else {
			canvas.drawCircle(coordinateX.toFloat(), coordinate.toFloat(), radius.toFloat(), paint)
		}
	}
}
